File: server/src/models/modifier.ts
import { model, Schema } from 'mongoose';
import type { ModifierDocument } from '../typings/models';

const modifierSchema = new Schema<ModifierDocument>({
  name: String,
  material: String,
  description: [String],
  enabled: Boolean,
});

export default model<ModifierDocument>('Modifier', modifierSchema, 'modifiers');

File: server/mongo/entities/modifier.go
package entities

type Modifier struct {
	Name        string   `bson:"name" json:"name"`
	Enabled     bool     `bson:"enabled" json:"enabled"`
	Material    string   `json:"material" bson:"material"`
	Description []string `json:"description" bson:"description"`
	Web         []string `json:"web,omitempty" bson:"web"`
}
